web/impact/impact/v0/views/utils.py

```diff
@@ -45,9 +45,20 @@ def _pad(text):
 
 
 def logo_url(startup):
-    if not startup.high_resolution_logo:
+    """
+    Turns the stored *.s3.amazonaws.com URL into one that uses our
+    CloudFormation image resizer; returns empty string for None
+    """
+    logo_field = startup.high_resolution_logo
+    if not logo_field:
         return ""
-    return startup.high_resolution_logo.url
+    elif "startup_pics/" not in logo_field.url:
+        return logo_field.url
+    else:
+        # Get just the filename and drop any S3 auth params
+        filename = logo_field.url.split("startup_pics/")[1].split("?")[0]
```

Comment:
much better to use logo_field.name instead of logo_field.url. It will save you the s3 params.

Also, the `startup_pics` part is required to make the path unique (within the MEDIA_ROOT). Not sure that splitting it out is a smart call, and I'm not sure where to look at to evaluate the other end of this decision.

Reply:
Thanks for the `logo_field.name` tip, that's perfect. I'll think about the `startup_pics` issue -- right now, this solution is handling those images and no others.